src/main/java/duke/commands/FindCommand.java

```diff
@@ -0,0 +1,41 @@
+package duke.commands;
+
+import duke.tasks.Task;
+import duke.tasklist.TaskList;
+import duke.ui.Ui;
+
+import java.util.ArrayList;
+import java.util.Locale;
+import java.util.stream.Collectors;
+
+public class FindCommand extends Command {
+    /**
+     * Find the list of tasks with matching keywords
+     *
+     * @param taskList list of task
+     * @param userInput user input
+     * @throws ArrayIndexOutOfBoundsException if no tasks found with given keyword
+     */
+    public FindCommand(TaskList taskList, String userInput) {
+        try {
+            String keyword = userInput.split(" ")[1];
+            ArrayList<Task> foundTasks = (ArrayList<Task>) taskList.getTasks().stream()
+                    .filter(t -> t.getDescription().toLowerCase(Locale.ROOT).contains(keyword))
+                    .collect(Collectors.toList());
+
+            boolean noMatch = foundTasks.isEmpty();
+
+            if (noMatch) {
+                Ui.printNoTaskFound();
+                return;
+            }
+
+            System.out.println("Here are the matching tasks in your list containing "
+                                + "\"" + keyword + "\"" + ":");
```

Comment:
Like the unique response from the bot!
I was wondering if you could shift this print statement to another class?
A suggestion would be the ui.java class.

Otherwise, excellent work!